interbtc-spec/docs/source/spec/nomination.rst

```diff
@@ -208,463 +131,154 @@ Specification
 
 optInToNomination
 -----------------
 
-Become an Operator in the Vault Nomination protocol
+Allow the Vault to receive nominated collateral.
 
 Specification
 .............
 
 *Function Signature*
 
-``optInToNomination(operatorId)``
+``optInToNomination(vaultId)``
 
 *Parameters*
 
-* ``operatorId``: the id of the vault to mark as Nomination Operator.
+* ``vaultId``: the id of the vault to enable nomination for.
 
 *Events*
 
-* ``NominationOptIn(operatorId)``
+* ``NominationOptIn(vaultId)``
 
-*Errors*
+*Preconditions*
 
-* ``VaultNominationDisabled``: the nomination feature is disabled.
-* ``NotAVault``: the caller of the function is not a vault.
-* ``VaultAlreadyOptedInToNomination``: the caller of the function is already opted in.
+* The BTC Parachain status in the :ref:`security` component MUST NOT be ``SHUTDOWN:2``.
+* A vault with id ``vaultId`` MUST be registered.
 
-Preconditions
-.............
-
-* The BTC Parachain status in the :ref:`security` component must be set to ``RUNNING:0``.
-
-
-Function Sequence
-.................
+*Postconditions*
 
-1. Check if the nomination feature is enabled. If not, throw ``VaultNominationDisabled``.
-
-2. Check if the caller is a vault. If not, throw ``NotAVault``.
-
-3. Check if the caller is not already opted in to nomination. If not, throw ``VaultAlreadyOptedInToNomination``.
-
-4. Instantiate an ``Operator`` struct.
-
-5. Add the struct to the ``Operators`` mapping.
+* The vault MUST be allowed to receive nominated collateral.
 
 
 .. _optOutOfNomination:
 
 optOutOfNomination
 ------------------
 
-Deregister from being an Operator in the Vault Nomination protocol.
+Disallow the Vault from receiving nominated collateral and force refund nominators.
 
 Specification
 .............
 
 *Function Signature*
 
-``optOutOfNomination(operatorId)``
+``optOutOfNomination(vaultId)``
 
 *Parameters*
 
-* ``operatorId``: the id of the vault to deregister from the nomination feature.
+* ``vaultId``: the id of the vault to deregister from the nomination feature.
 
 *Events*
 
-* ``NominationOptOut(operatorId)``
-
-*Errors*
+* ``NominationOptOut(vaultId)``
 
-* ``VaultNotOptedInToNomination``: the caller is not an Operator.
+*Preconditions*
 
-Preconditions
-.............
-
-* The BTC Parachain status in the :ref:`security` component must be set to ``RUNNING:0``.
-
-
-Function Sequence
-.................
+* The BTC Parachain status in the :ref:`security` component MUST NOT be ``SHUTDOWN:2``.
+* A vault with id ``vaultId`` MUST be registered.
+* A vault with id ``vaultId`` MUST have nomination enabled.
 
-1. Check if the caller is a nomination Operator. If not, throw ``VaultNotOptedInToNomination``.
+*Postconditions*
 
-2. Immediately refund all nominated collateral, bypassing the unbonding period.
+* The vault MUST NOT be allowed to receive nominated collateral.
+* Previous nominators MUST NOT be allowed to receive more rewards.
+* Previous nominators MUST be allowed to withdraw their collateral.
```

Comment:
This is quite vague and hard to check. I think it's better to be concrete, rather than having the 3 points above. E.g. just say that the vault must be removed from the vaults mapping